glusterd2/daemon/daemon.go

```diff
@@ -73,15 +75,18 @@ func Start(d Daemon, wait bool) error {
 
 	args := strings.Fields(d.Args())
 	cmd := exec.Command(d.Path(), args...)
-	err = cmd.Start()
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+
+	err = utils.ExecuteCommandStart(cmd, &stderr)
 	if err != nil {
 		events.Broadcast(newEvent(d, daemonStartFailed, 0))
 		return err
 	}
 
 	if wait == true {
 		// Wait for the child to exit
-		errStatus := cmd.Wait()
+		errStatus := utils.ExecuteCommandWait(cmd, &stderr)
```

Comment:
The daemon framework will only run various daemons which don't have anything attached to it's STDOUT and STDERR. This change isn't necessary for daemon framework. Further, in this specific case, it's easier to read and understand `cmd.Wait()`